95.md

```diff
@@ -0,0 +1,75 @@
+NIP-95 - Storage and Shared File
+======
+`draft` `optional` `author:frbitten` 
+
+This NIP allows users to send binary data to be stored in the relay. This data can be small pieces of information or even large files.
+
+The intention is to combine the ease of NOSTR, with many clients being developed and a common communication protocol between all. To enable decentralized file sharing. NOSTR communication must work together with other protocols to get the best out of each technology. For lay users to be able to publish their photo, file, audio on several servers in a simple and easy way.
+
+Nostr event
+------------------
+This NIP specifies the use of the `1064` event type (parameterized replaceable event), having in `content` the binary data that you want to be stored in Base64 format.
+* `type` a string indicating the data type of the file. The MIME types format must be used (https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types)
+* `decrypt` (optional) If the file is encrypted, you must indicate in the first value the algorithm used to encrypt the file and in the second value the parameters that allow the decryption of the file
+
+This event must not be returned in generic searches. It should only be returned if specifically requested by your ID. This avoids bandwidth consumption and unnecessary overload on the relay and client.
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded sha256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <unix timestamp in seconds>,
+  "kind": 1064,
+  "tags": [
+    ["decrypt",<algorithm>,<Decryption Params>],
+    ["p", <32-bytes hex of a pubkey>, <recommended relay URL>],
```

Comment:
Is `p` redundant with `pubkey` here? Is this just so that you can recommend a relay URL for that pubkey?

Do you want to include an example `type` tag here?  Actually, metadata like that should probably be moved to the 1065 kind, right? Would be useful for clients to know mime type, size, dimensions, etc, before choosing whether to fetch the file.

Reply:
Yes. Just to reference other users, following the NIP-01 standard. It is not mandatory.